### DISCUSSION TITLE
Usability Issue in Joomla when filter text

### DISCUSSION THREAD

**Discussion by B3nito** [joomla/joomla-cms#47143] [2. Cold Features]
### Is your feature request related to a problem? Please describe.
When you are a enduser and use an editor and plugin settings is filtering input then html is just stripping out of text.

### Describe the solution you'd like
User should get notification that html was cleaning for plugin settings so better usability

### Additional context
I have problem in custom fields that filter is enabled default and user thought that Joomla is buggy. User should know what system is do automatically.

**Comment by Ayushsunny** [comment]
Is this issue still persist?

**Comment by pank1999** [comment]
i want to solve this issue please guide me

**Comment by Nitesh639** [comment]
I want to solve this issue.

**Comment by Yash-g17** [comment]
@HLeithner I'd love to work on this issue , can you please assign and guide me ?

**Comment by brianteeman** [comment]
The Joomla Project doesn't assign issues to contributors in that way. Just work on it and submit a PR :-)

https://github.com/joomla/joomla-cms/blob/staging/.github/CONTRIBUTING.md

**Comment by Ayushsunny** [comment]
> The Joomla Project doesn't assign issues to contributors in that way. Just work on it and submit a PR :-)
> 
> https://github.com/joomla/joomla-cms/blob/staging/.github/CONTRIBUTING.md

is there any mentor who can help us to solve this issue?

**Comment by abhijha0602** [comment]
I want to solve this issue

**Comment by ghost** [comment]
> Just work on it and submit a PR :-)

**Comment by imtiyaj786** [comment]
how can i check this issue in own ?

**Comment by ReLater** [comment]
Hello volunteers,
- You don't have to ask if you are allowed to take over an issue.
- - Identify the issue in an installation of Joomla. Use a nightly build for it: https://developer.joomla.org/nightly-builds.html
- - Try to find a code solution for the issue after you have checked whether it really exists.
- - If it exists confirm it here in the issue for the sake of completeness. A "I'll take over" is a good hint for others that don't waste their time then with it.
- - Test your solution.
- - Finally provide a pull request on GitHub (= "here") that others can test. If tests are not successful you can change and discuss your pull request for further tests.
- - Sometimes others provide a "suggestion" that you can merge directly by clicking in your PR.
- There are no mentors to accompany you. It goes without saying that you can ask the general public how this or that works. I think (don't know) that the "Discussions" tab above can be a good place for more general questions.
- Here is a doc how to create pull requests via browser GUI: https://docs.joomla.org/Using_the_Github_UI_to_Make_Pull_Requests 
- If you want to go deeper you could also work with a git repository tool on your local machine. Needs some more skills and time to learn:

**Comment by amankothiyal04** [comment]
hey @B3nito can you help me with this issue want to contribute to solve it

**Comment by Shaveetha** [comment]
I would like to solve this issue. Can you please guide me.

**Comment by brianteeman** [comment]
@Shaveetha @amankothiyal04 

> Hello volunteers,
> 
> * You don't have to ask if you are allowed to take over an issue.
> * * Identify the issue in an installation of Joomla. Use a nightly build for it: https://developer.joomla.org/nightly-builds.html
> * * Try to find a code solution for the issue after you have checked whether it really exists.
> * * If it exists confirm it here in the issue for the sake of completeness. A "I'll take over" is a good hint for others that don't waste their time then with it.
> * * Test your solution.
> * * Finally provide a pull request on GitHub (= "here") that others can test. If tests are not successful you can change and discuss your pull request for further tests.
> * * Sometimes others provide a "suggestion" that you can merge directly by clicking in your PR.
> * There are no mentors to accompany you. It goes without saying that you can ask the general public how this or that works. I think (don't know) that the "Discussions" tab above can be a good place for more general questions.
> * Here is a doc how to create pull requests via browser GUI: https://docs.joomla.org/Using_the_Github_UI_to_Make_Pull_Requests
> * If you want to go deeper you could also work with a git repository tool on your local machine. Needs some more skills and time to learn:

**Comment by brianteeman** [comment]
Can you please remove the "good first issue label" as its absolutely not a simple thing to fix (not sure if its even possible)

**Comment by HLeithner** [comment]
I don't think it's too hard.

I would solve this issue with 5 lines of code (without looking in deep, only looked at com_content)

Adding `$original = $text` at this position: https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Component/ComponentHelper.php#L126

And a `if ($text !== $original) { JFactory::getApplication()->enqueueMessage('JHTML_FILTER_MODIFY', CMSApplicationInterface::MSG_WARNING); }` at line https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Component/ComponentHelper.php#L295

Of course the language string have to be added too. But maybe I think too simple and missed something ymmv

**Comment by brianteeman** [comment]
1. That won't work for text filtering where the editor eg tinymce is doing the filtering
2. To be useful it really should say what has been filtered

**Comment by HLeithner** [comment]
2. is not possible (at least I can't think of a working solution)

1. I'm pretty sure there is a way to achieve this for tinymce in js too

**Comment by brianteeman** [comment]
Its still not a "good first issue" which was the point of my post

**Comment by HLeithner** [comment]
Why not? it's easy to fix, the first parts are only 5 lines of code.

What would you say what's a good first issue is, If you find one I would really like to tag it.

**Comment by brianteeman** [comment]
If it was so easy then any one of the 10 people who said they wanted to fix it would have been able to or anyone else for that matter in the months since november. plus if its really so easy then you would have submitted it instead of debating a pointless argument with me.

**Comment by teekhandekar** [comment]
@brianteeman Can I work on this ?

**Comment by adithyareddych** [comment]
@brianteeman Can I work on this issue?

**Comment by anshumaan255** [comment]
is this issue still open want to work on it

**Comment by ghost** [comment]
@anshumaan255 https://github.com/joomla/joomla-cms/issues/31540#issuecomment-811070565

**Comment by dv2202** [comment]
Hey,@B3nito 
Can i work on this issue ?

**Comment by joomdonation** [comment]
@dv2202 No need to ask. See https://github.com/joomla/joomla-cms/issues/31540#issuecomment-811070565 for more details.

**Comment by YashSalve695** [comment]
can you please assign me this issue?

**Comment by kushvahasumit** [comment]
i also want to contribute for this one.

**Comment by farmannaseem** [comment]
**_I wanna work on this issue please assign me this issue i would like to contribute in it_**
